lewis/adapters/stream.py

```diff
@@ -86,7 +84,44 @@ def handle_accept(self):
             StreamHandler(sock, self.target)
 
 
-class Cmd(object):
+class CanNotProcessException(Exception):
+    pass
```

Comment:
Thought about this some more. It's a bit awkward no matter how you do it. But I'm still leaning towards the tuple return.

Introducing `can_process` would give us a way to check before calling `process_request`, but then you have to use the same match in both places. And I could just call `process_request` with garbage values anyway... so it would still need to throw an exception on no match. So, in effect, we haven't simplified anything and only added code duplication and complexity.

Chain of responsibility seems like overkill for this simple situation. We could consider a simplified version, like what the `re` module does: Take parameters normally but return a `match` object that may or may not have results, and contains meta information about the nature and validity of results.

But what would our return object need to look like / what values would it contain? It would contain only two things: 

1. A boolean to indicate whether the request was accepted/processed, and 
2. A return value from processing the request (default `None` if above `False`). 

So: `(False, None)` or `(True, ret_value)`. Why create a new type for this if tuple fits the bill?